packages/astro/src/compiler/transform/styles.ts

```diff
@@ -125,7 +125,7 @@ async function transformStyle(code: string, { logging, type, filename, scopedCla
     try {
       const require = createRequire(import.meta.url);
       const tw = require.resolve('tailwindcss', { paths: [import.meta.url, process.cwd()] });
-      postcssPlugins.push(require(tw) as any);
+      postcssPlugins.push(require(tw)(tailwindConfig) as any);
```

Comment:
Same here—I think we’re trying to load the PostCSS plugin with a string, rather than a JSON object. I believe we need to load the user’s tailwind config first?

Reply:
Actually, on second thought, I just found this in [Tailwind’s Docs](https://tailwindcss.com/docs/configuration#using-a-different-file-name). Maybe this would work instead?

```suggestion
-     postcssPlugins.push(require(tw)(tailwindConfig) as any);
+    postcssPlugins.push(require(tw)({ config: tailwindConfig }) as any);
```

But please run `yarn build` and test the `examples/tailwindcss` example to make sure that this change is working (you can replicate your project in there to test it).

Reply:
Hey @drwpow. There are a few different ways you can pass a config to Tailwind. The way I have used works fine but I can change it to the `{ config }` format if you prefer?

Reply:
Ah no, you’re right this does seem to be working in your branch. I apologize—I tried testing it earlier and it wasn’t working, but I think something unrelated was interfering. Keep as-is!